akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -242,7 +242,8 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
     settings:          ServerSettings    = ServerSettings(system),
     parallelism:       Int               = 1,
     log:               LoggingAdapter    = system.log)(implicit fm: Materializer): Future[ServerBinding] =
-    bindAndHandle(Flow[HttpRequest].mapAsync(parallelism)(handler), interface, port, connectionContext, settings, log)
+    if (settings.useHttp2) Http2Shadow.bindAndHandleAsync(handler, interface, port, connectionContext, settings, parallelism, log)(fm)
```

Comment:
Oh, good idea - current impl is a bit aggresive and blows up if it's not, falling back is nicer indeed :)